daveengelberg-smangelberg/robots/robot_haus.rb

```diff
@@ -0,0 +1,129 @@
+# instance thing. We basically just want to make sure that if we call
+# one of the instance methods on one of the robots created, we can handle whether
+# it does or doesn't have the attribute.
+
+require 'pry'
+require 'pry-byebug'
+
+class Robot
+
+  #creates getter && setter methods for each attribute
+  attr_accessor :name, :type, :origin
+
+ #instance_factory to support creating a new robot using Robot.new
+  def initialize(name, type,origin)
+    @name = name
+    @type = type
+    @origin = origin
+  end
+
+  def self.random_robot_maker
+    #randomly select name, type, origin, from three different arrays
+    name = ["Destructoid", "Kitten4000", "Hooman"].sample
+    type = ["four wheel","two wheel", "upright"].sample
+    origin = ["Belize", "China", "Canada"].sample
+    robot = Robot.new(name, type, origin)
+    return robot
+  end
+
+  #depending on type of robot, they will have different attributes
+  #instance methods can only be called by instances of a class
+  def empathy
+    print "Calling empathy"
+    2.times do
+      sleep(0.25)
+      print "."
+      sleep(0.25)
+    end
+    puts "."
+    sleep(0.25)
+    if @name != "Destructoid"
+      puts "SKILL DEPLOY: *** COMMENCING HUG ***"
+    else
+      puts "SKILL DEPLOY: No hugs bro!"
+    end
+  end
+
+  #this is an instance method
+  def kitten_skills
+    print "Calling kitten_skills"
+    2.times do
+      sleep(0.25)
+      print "."
+      sleep(0.25)
+    end
+    puts "."
+    sleep(0.25)
+    if @name == "Kitten4000"
+      puts "SKILL DEPLOY: Kitten town!"
+    else
+      puts "SKILL DEPLOY: Dogs for you"
+    end
+  end
+
+  def fourty_mph_skills
+    print "Calling Fourty_mph_skills"
+    2.times do
+      sleep(0.25)
+      print "."
+      sleep(0.25)
+    end
+    puts "."
+    sleep(0.25)
+    if @type = "four wheel" || @type = "two wheel"
+      puts "SKILL DEPLOY: Your robot can move pretty fast."
+    else
+      puts "SKILL DEPLOY: No dice."
+    end
+  end
+
+end
+
+# This is the start of the program. Based on my understanding
+# I don't think you want to combine a class for creating robots with
+# a class for how many robots you want to create. Better to have a class
+# that creates robots and then let the user create how many they need.
+# Also I couldn't figure out how to get the array working within the class.
+
+# Ask user how many bots they want.
+puts "Welcome to the Robot Haus. How many robots do you need?"
+number_of_robots_requested = gets.chomp.to_i
+
+# creating an empty array that includes all robots that are created, so they
+# could be queried
+robot_array = []
+
+# Loop through random_robot_maker
+number_of_robots_requested.times do
+  robot_array.push(Robot.random_robot_maker)
+end
+
+
+# this is a bullshit function. definitely should put some stuff into an
+# array and iterate through it. So much repeated code. But anyway,
+# it loops through each robot and calls each skill.
```

Comment:
It's okay to organize your code like this, even if it isn't super DRY. It's better to write concise code, yes, but readability has higher priority. There are plenty of clever one-liners out there, but that's no fun for others to figure out the inner workings.

You've kept your code both clean and well documented which is awesome. Keep it up, Dave! :octocat: